components/content-service/pkg/storage/storage.go

```diff
@@ -68,6 +69,15 @@ type PresignedAccess interface {
 
 	// DeleteObject deletes objects in the given bucket specified by the given query
 	DeleteObject(ctx context.Context, bucket string, query *DeleteObjectQuery) error
+
+	// DeleteBucket deletes a bucket
+	DeleteBucket(ctx context.Context, bucket string) error
+
+	// ObjectHash gets a hash value of an object
+	ObjectHash(ctx context.Context, bucket string, obj string) (string, error)
+
+	// BackupObject returns a backup's object name that a direct downloader would download
+	BackupObject(workspaceID string, name string) string
```

Comment:
Instead of introducing a second `BackupObject` function, it might make sense to adapt the `BackupObjectNamer` interface and include that one in `PresignedAccess`

Reply:
Absolutely agree.

The problem is, that we need another signature here. In general, I think we should completely rework the storage interface. The separation of `PresignedAccess` and `DirectAccess` makes no sense at all anymore in my opinion. This only leads us to duplicate code unnecessarily. I had already changed the naming a bit last time, but we took that back for the time being. Maybe this is material for a follow-up task?

Reply:
Could you create a follow-up issue that outlines the changes you'd make?

Reply:
→ #3437